### PR TITLE
Prevent onCancel if onClick Fails

### PR DESCRIPTION
**Pull request by mnicpt**
https://engineering.paypalcorp.com/jira/browse/DTCHKINT-867

### CODE REVIEW

File: src/payment-flows/native/native.js [hunk 1]
```diff
@@ -274,10 +274,9 @@ function initNative({ props, components, config, payment, serviceData } : InitOp
     const onCloseCallback = () => {
         return ZalgoPromise.delay(1000).then(() => {
             if (!approved && !cancelled && !didFallback && !isAndroidChrome()) {
-                return ZalgoPromise.all([
-                    onCancel(),
-                    destroy()
-                ]);
+                return ZalgoPromise.try(() => {
```
Comment (gregjopa) [inline src/payment-flows/native/native.js:277]: Why doesn't native need to call `onCancel` when an orderID is already created?

**Review by gregjopa — APPROVED** [review]
Looks good to me! I left one question about the native flow.

File: src/payment-flows/native/native.js (hunk 1 above)
Comment (mnicpt) [reply src/payment-flows/native/native.js:277]: This callback occurs when popup is closed either manually or from false being returned from `onClick`.  In each of these cases we don't want to call `onCancel`.  Closing manually follows same pattern as checkout flow.

File: src/payment-flows/checkout.js [hunk 2]
```diff
@@ -241,6 +241,7 @@ function initCheckout({ props, components, serviceData, payment, config } : Init
     const connectEligible = isConnectEligible({ connect, createBillingAgreement, createSubscription, vault, fundingSource });
 
     let approved = false;
+    let orderCreated = false;
```
Comment (bluepnume) [inline src/payment-flows/checkout.js:244]: I think we don't need these changes in `checkout.js`. `start()` should not even be called if `onClick()` returns false so this should never happen.

**Review by bluepnume — APPROVED** [review]

**Pull request merged**